fujiji-server/__test__/controllers/listing.test.js

```diff
@@ -22,5 +22,61 @@ describe('Test /listing endpoints', () => {
         .get('/listing');
       expect(res.statusCode).toEqual(400);
     });
+
+    it('successfully get all listings by existing city and category', async () => {
+      const res = await request(app)
+        .get('/listing')
+        .query({ city: 'Winnipeg', category: 'Other' });
+      expect(res.statusCode).toEqual(200);
+    });
+
+    it('successfully get all listings by existing province and category', async () => {
+      const res = await request(app)
+        .get('/listing')
+        .query({ provinceCode: 'MB', category: 'Other' });
+      expect(res.statusCode).toEqual(200);
+    });
+
+    it('successfully get all listings by existing city and condition', async () => {
+      const res = await request(app)
+        .get('/listing')
+        .query({ city: 'Winnipeg', condition: 'used' });
+      expect(res.statusCode).toEqual(200);
+    });
+
+    it('successfully get all listings by existing province and condition', async () => {
+      const res = await request(app)
+        .get('/listing')
+        .query({ provinceCode: 'MB', category: 'used' });
+      expect(res.statusCode).toEqual(200);
+    });
+
+    it('successfully get all listings by existing city and valid priceRange', async () => {
+      const res = await request(app)
+        .get('/listing')
+        .query({ city: 'Winnipeg', priceRange: [0, 100] });
+      expect(res.statusCode).toEqual(200);
+    });
+
+    it('successfully get all listings by existing province and valid priceRange', async () => {
+      const res = await request(app)
+        .get('/listing')
+        .query({ provinceCode: 'MB', priceRange: [0, 100] });
+      expect(res.statusCode).toEqual(200);
+    });
+
+    it('invalid price range with existing city returns expected error', async () => {
+      const res = await request(app)
+        .get('/listing')
+        .query({ city: 'Winnipeg', priceRange: [100] });
+      expect(res.statusCode).toEqual(400);
+    });
+
+    it('invalid price range with existing province returns expected error', async () => {
+      const res = await request(app)
+        .get('/listing')
+        .query({ provinceCode: 'MB', priceRange: [100] });
+      expect(res.statusCode).toEqual(400);
+    });
```

Comment:
We probably need refactor this tests after we setup DB for testing 😭

e.g., testing if we get the actual listings

Otherwise, lgtm